src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,136 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.regex.Pattern;
+
+public class Parser {
+    TaskList taskList;
+    private static final Pattern DATE_PATTERN_DASH = Pattern.compile("^\\d{4}-\\d{2}-\\d{2}$");
+    private static final Pattern DATE_PATTERN_SLASH = Pattern.compile("^\\d{1,2}/\\d{1,2}/\\d{4} .*\\d{4}$");
+    public Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Writes task into Local Storage.
+     * @param promptText    Task from user's input.
+     * @throws DukeException    Exception to be thrown when the input cannot be read.
+     */
+    public void createTask(String promptText) throws DukeException {
+        if (promptText.startsWith("todo")) {
+            try {
+                Task todo = new Todo(promptText.substring(5));
+                taskList.add(todo, true);
+                taskList.writeToFile();
+            }
+            catch (StringIndexOutOfBoundsException s) {
+                throw new DukeException("OOPS!! The description of a todo cannot be empty.");
+            }
+        }
+        else if (promptText.startsWith("deadline")) {
+            try {
+                String[] parts = promptText.split("/", 2);
```

Comment:
Perhaps you could use a more descriptive variable name to help with understanding.